### PR TITLE
Fix Sonarcloud errors

### PR DESCRIPTION
**Pull request by sebr72**

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/map/geotools/grid/GridLayer.java [hunk 1]
```diff
@@ -134,7 +121,26 @@ public void render(
...
+          try {
+              return new Font(
+                  fontName,
+                  this.params.font.style.styleId,
+                  (int) (this.params.font.size * dpiScaling));
+          } catch (RuntimeException e) {
+            // try next font in list
+          }
+        }
+        throw new NullPointerException("Found no matching font for dpiScaling:" + dpiScaling);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/geotools/grid/GridLayer.java:140]: You just add a NullPointerException?
Why does it fix a SonarCloud issue?
Shouldn't be a RuntimeException

File: core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java [hunk 2]
```diff
@@ -53,32 +53,50 @@ public PArray getArray(final int i) {
...
+
+  /**
+   * Created to keep creating a controlled {@link NullPointerException}.
+   * @param index in the array
+   * @return the object or an NPE if index is out of bounds.
+   */
+  private Object getObjectAt(final int index) {
+    Object o = get(index);
+    if (o == null) {
+      throw new NullPointerException("No value found at index:" + index);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java:97]: I didn't like to raise a NullPointerException...

File: core/src/main/java/org/mapfish/print/map/geotools/grid/GridLayer.java (hunk 1 above)
Comment (sebr72) [reply core/src/main/java/org/mapfish/print/map/geotools/grid/GridLayer.java:140]: New RuntimeException as discussed

File: core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java (hunk 2 above)
Comment (sebr72) [reply core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java:97]: New RuntimeException as discussed

**Review by sbrunner — APPROVED** [review]

**Pull request merged**